codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/protocols/restjson/AWSRestJson1ProtocolGenerator.kt

```diff
@@ -14,6 +14,7 @@ class AWSRestJson1ProtocolGenerator : AWSHTTPBindingProtocolGenerator(RestJSONCu
     override val testsToIgnore = setOf(
         "SDKAppliedContentEncoding_restJson1",
         "SDKAppendedGzipAfterProvidedEncoding_restJson1",
+        // This test is disabled until Smithy v1.51.0 with the fix for the test gets released.
+        "RestJsonClientIgnoresDefaultValuesIfMemberValuesArePresentInResponse"
```

Comment:
Please file an issue in the SDK project to track this

Reply:
Created the issue for follow-up: https://github.com/awslabs/aws-sdk-swift/issues/1706